fix(router-core): change error type in ErrorComponentProps

leesb971204 opened the pull request:
fixes #4686

leesb971204 commented on packages/router-core/src/route.ts:
```diff
@@ -1318,7 +1318,7 @@ export type ErrorRouteProps = {
 }
 
 export type ErrorComponentProps = {
-  error: Error
+  error: unknown
```
or like [axiosError type](https://github.com/axios/axios/blob/8517aa16f8d082fc1d5309c642220fa736159110/index.d.ts#L405C1-L405C60)?
```tsx
export type ErrorComponentProps<TError extends Error> = {
  error: TError
```

leesb971204 replied:
@michaelbull 

Thank you for the feedback!
Could you please share more details about the approach you mentioned from React Query? Perhaps an issue or PR?

From what I can see in the official [React Query documentation](https://tanstack.com/query/latest/docs/framework/react/typescript#registering-a-global-error), it still seems to be using `declare` for now.

leesb971204 commented on packages/router-core/src/route.ts:
```diff
@@ -1317,8 +1317,8 @@ export type ErrorRouteProps = {
 
-export type ErrorComponentProps = {
-  error: Error
+export type ErrorComponentProps<TError = Error> = {
+  error: TError
```
This reflects @michaelbull ’s suggestion.

The change respects the diversity of error objects returned from the backend while setting `Error` as the default type to ensure compatibility with existing code.

schiller-manuel commented:
thanks for this PR!

can you please update the docs 

docs/router/framework/react/api/router/errorComponentComponent.md

leesb971204 commented:
> thanks for this PR!
> 
> can you please update the docs
> 
> docs/router/framework/react/api/router/errorComponentComponent.md

I updated the type of the `error`, and also added a description for `info`, which seemed to be missing.
